asset_stream_manager/start_command.cc

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "asset_stream_manager/start_command.h"
+
+#include <memory>
+
+#include "asset_stream_manager/local_assets_stream_manager_client.h"
+#include "asset_stream_manager/session_management_server.h"
+#include "common/log.h"
+#include "common/path.h"
+#include "common/status_macros.h"
+#include "lyra/lyra.hpp"
+
+namespace cdc_ft {
+
+StartCommand::StartCommand(int* exit_code)
+    : BaseCommand("start",
+                  "Start streaming files from a Windows to a Linux device",
+                  exit_code) {}
+
+StartCommand::~StartCommand() = default;
+
+void StartCommand::RegisterCommandLineFlags(lyra::command& cmd) {
+  verbosity_ = 2;
+  cmd.add_argument(lyra::opt(verbosity_, "num")
+                       .name("--verbosity")
+                       .help("Verbosity of the log output, default: " +
+                             std::to_string(verbosity_) +
+                             ".Increase to make logs more verbose."));
+
+  service_port_ = SessionManagementServer::kDefaultServicePort;
+  cmd.add_argument(lyra::opt(service_port_, "port")
+                       .name("--service-port")
+                       .help("Local port to use while connecting to the local "
+                             "asset stream service, default: " +
+                             std::to_string(service_port_)));
+
+  ssh_port_ = 22;
+  cmd.add_argument(
+      lyra::opt(ssh_port_, "port")
+          .name("--ssh-port")
+          .help("Port to use while connecting to the remote instance being "
+                "streamed to, default: " +
+                std::to_string(ssh_port_)));
+
+  path::GetEnv("CDC_SSH_COMMAND", &ssh_command_).IgnoreError();
```

Comment:
What would happen if there is a variable (for example, outdated from previous runs) and the user defines a new one. Which will survive?

Reply:
Not sure what you mean. Commands only run once, so there can't be a "previous" ssh_command_.

Reply:
After talking offline, if there's an environment variable set and a command line flag, the command line flag will win.